tracer/tracer.go

```diff
@@ -284,11 +284,24 @@ func initializeMapsAndPrograms(kmod *kallsyms.Module, cfg *Config) (
 	// References to eBPF maps in the eBPF programs are just placeholders that need to be
 	// replaced by the actual loaded maps later on with RewriteMaps before loading the
 	// programs into the kernel.
+	major, minor, patch, err := GetCurrentKernelVersion()
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get kernel version: %v", err)
+	}
+
 	coll, err := support.LoadCollectionSpec()
 	if err != nil {
 		return nil, nil, fmt.Errorf("failed to load specification for tracers: %v", err)
 	}
 
+	if major >= 6 && minor >= 16 {
+		// Tracepoint format for sched_process_free has changed in v6.16+.
+		// We therefore use two different hooks and decide which to load at runtime.
+		delete(coll.Programs, "tracepoint__sched_process_free_old")
+	} else {
+		delete(coll.Programs, "tracepoint__sched_process_free")
```

Comment:
This determines also the tracepoint name to use. Could it be passed to `loadPerfUnwinders` and returned back to `Tracer` constructor, so the name is determined once and then used consistently?